src/connector/src/sink/file_sink/opendal_sink.rs

```diff
@@ -331,3 +531,48 @@ fn convert_rw_schema_to_arrow_schema(
 
     Ok(arrow_schema_iceberg::Schema::new(arrow_fields))
 }
+
+/// `BatchingStrategy` represents the strategy for batching data before writing to files.
+///
+/// This struct contains settings that control how data is collected and
+/// partitioned based on specified criteria:
+///
+/// - `max_row_count`: Optional maximum number of rows to accumulate before writing.
+/// - `rollover_seconds`: Optional time interval (in seconds) to trigger a write,
+///   regardless of the number of accumulated rows.
+/// - `path_partition_prefix`: Specifies how files are organized into directories
+///   based on creation time (e.g., by day, month, or hour).
+
+#[serde_as]
+#[derive(Default, Deserialize, Debug, Clone, WithOptions)]
+pub struct BatchingStrategy {
+    #[serde(default)]
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub max_row_count: Option<usize>,
+    #[serde(default)]
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub rollover_seconds: Option<usize>,
+    #[serde(default)]
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub path_partition_prefix: Option<PathPartitionPrefix>,
+}
+
+/// `PathPartitionPrefix` defines the granularity of file partitions based on creation time.
+///
+/// Each variant specifies how files are organized into directories:
+/// - `None`: No partitioning.
+/// - `Day`: Files are written in a directory for each day.
+/// - `Month`: Files are written in a directory for each month.
+/// - `Hour`: Files are written in a directory for each hour.
```

Comment:
I think we need to consider timezone here. Either we force it to be UTC or we expose a config for the user to specify the desired timezone.

Reply:
Let's first force it to UTC and indicate this in the documentation. Because partitioning is originally a niche demand, we can make it configurable when users need other time zones.